### ISSUE TITLE
SCT does not respect arch for scylla ami's on 2021.1 branch

### ISSUE THREAD

**Issue by soyacz** [scylladb/scylla-cluster-tests#5377]
An upgrade test for 2021.1 failed due to taking up ARM arch based AMI on x86_64 machines: https://jenkins.scylladb.com/view/nexts/job/enterprise-2021.1/job/rolling-upgrade/job/rolling-upgrade-ami-test/107

Looks we didn't backport arch related features to 2021.1 branch (do we support ARM for 2021.1?).

**Comment by soyacz** [comment]
We could try fixing it in 2 ways:
1. backport several related commits (could be hard)
2. take master's `get_scylla_ami_versions`, `get_arch_from_instance_type` functions content and copy them to 2021.1 branch (should be easier, with less risk)
@fruch @roydahan @fgelcer what do you think?

**Comment by fruch** [comment]
Do we have arm images for 2021.1 ?

**Comment by soyacz** [comment]
`ami-0c3403b5165bd1d02` which is `ScyllaDB Enterprise 2021.1.15 ARM`

**Comment by fruch** [comment]
> `ami-0c3403b5165bd1d02` which is `ScyllaDB Enterprise 2021.1.15 ARM`

yeah, seems like since 2021.10 we have ARM images

```
+---------+-----------------------------------+-----------------------+--------------------------+
| Backend | Name                              | ImageId               | CreationDate             |
+---------+-----------------------------------+-----------------------+--------------------------+
| AWS     | ScyllaDB Enterprise 2021.1.15 ARM | ami-0c3403b5165bd1d02 | 2022-10-13T10:17:19.000Z |
| AWS     | ScyllaDB Enterprise 2021.1.14 ARM | ami-069b735d4a25c59d0 | 2022-08-29T13:03:36.000Z |
| AWS     | ScyllaDB Enterprise 2021.1.13 ARM | ami-009e54fa40b6e2a79 | 2022-08-18T12:32:42.000Z |
| AWS     | ScyllaDB Enterprise 2021.1.12 ARM | ami-0f555314cce80e2ac | 2022-06-26T11:25:28.000Z |
| AWS     | ScyllaDB Enterprise 2021.1.11 ARM | ami-0b59381a0febd2352 | 2022-05-30T11:55:46.000Z |
| AWS     | ScyllaDB Enterprise 2021.1.10 ARM | ami-05c9c76f32caff625 | 2022-04-12T13:27:04.000Z |
+---------+-----------------------------------+-----------------------+--------------------------+
```

I would argue to first try option 1 (but in a separate PR), and only if it proves to be too problematic resort to option 2.

**Comment by fruch** [comment]
I think those might be the things we need backported:
https://github.com/scylladb/scylla-cluster-tests/pull/4326
https://github.com/scylladb/scylla-cluster-tests/pull/4321

**Comment by fgelcer** [comment]
> I think those might be the things we need backported: #4326 #4321

@fruch , cherry-pick doesn't apply cleanly.. one of the files in these 2 PRs doesn't even exist in `branch-2021.1`
wouldn't it be easier to just skip if `arch` label is `ARM` and go to the next one in the list of results?

**Comment by fruch** [comment]
It's depends what you want to achieve, I was under the assumption you want to be able to run ARM based upgrade test...

if it's only to filter out the ARM, should be easy enough to patch

**Comment by fgelcer** [comment]
> It's depends what you want to achieve, I was under the assumption you want to be able to run ARM based upgrade test...
> 
> if it's only to filter out the ARM, should be easy enough to patch

AFAICT formally 2021.1 doesn't support ARM, although since 2021.1.7 (we saw it from 2021.1.10) we build ARM instances, but it wasn't tested, hence not formally supported for that branch.
IIUC we want now only to filter ARM instances out, to avoid what we saw in that rolling upgrade, that when starting from latest released 2021.1.x, we found and used an ARM instance, instead of the expected x86_64

what do you suggest? query out ARM? `grep -v` ?

**Comment by fruch** [comment]
> > It's depends what you want to achieve, I was under the assumption you want to be able to run ARM based upgrade test...
> > 
> > if it's only to filter out the ARM, should be easy enough to patch
> 
> AFAICT formally 2021.1 doesn't support ARM, although since 2021.1.7 (we saw it from 2021.1.10) we build ARM instances, but it wasn't tested, hence not formally supported for that branch.
> IIUC we want now only to filter ARM instances out, to avoid what we saw in that rolling upgrade, that when starting from latest released 2021.1.x, we found and used an ARM instance, instead of the expected x86_64
> 
> what do you suggest? query out ARM? `grep -v` ?

It's not exactly grep, but we just need to fix the filter, and that would stop us from getting arm amis on this branch.

If we ever want to test those images on this branch it would need a bit more work getting the needed part into place.

Any how I found it a bit disturbing that we release those images and not testing them at all.

**Comment by roydahan** [comment]
I don’t know why we’re building ARM AMIs for this release.

**Comment by fgelcer** [comment]
> I don’t know why we’re building ARM AMIs for this release.

we cleared it out with @yaronkaikov .. now we need to decide who will address it on SCT

**Comment by fruch** [comment]
> > I don’t know why we’re building ARM AMIs for this release.
> 
> we cleared it out with @yaronkaikov .. now we need to decide who will address it on SCT

cleared as in ? arm images should be officially supported for 2021.1 or not ?

**Comment by yaronkaikov** [comment]
> > > I don’t know why we’re building ARM AMIs for this release.
> > 
> > 
> > we cleared it out with @yaronkaikov .. now we need to decide who will address it on SCT
> 
> cleared as in ? arm images should be officially supported for 2021.1 or not ?

Yes we do

**Comment by fgelcer** [comment]
@fruch @roydahan let's please prioritize it, as we may face it again in the next iterations of 5.1

**Comment by fruch** [comment]
this should be enough to patch it
https://github.com/scylladb/scylla-cluster-tests/pull/5471

if anyone has spare cycles to backport the ARM support... be my guest

**Comment by fruch** [comment]
Fixed by #5471
